src/CrudTrait.php

```diff
@@ -22,11 +22,14 @@ public static function getPossibleEnumValues($field_name)
 
         $instance = new static(); // create an instance of the model to be able to get the table name
         $connectionName = $instance->getConnectionName();
-        $type = DB::connection($connectionName)->select(DB::raw('SHOW COLUMNS FROM `'.$table_prefix.$instance->getTable().'` WHERE Field = "'.$field_name.'"'))[0]->Type;
+        $type = \DB::connection($connectionName)->getSchemaBuilder()->getColumnType( $table_prefix.$instance->getTable(), $field_name );
```

Comment:
I tested this on MySQL and it doesn't seem to work for me. Before this change, ```$type``` was ```enum('PUBLISHED', 'UNPUBLISHED')```. After this change, it's always ```string```, even if the db column is actually enum. I double-checked. I think it might be because of the ```CrudTrait::isColumnNullable()``` method, or somewhere else where the ```enum``` columns are defined to be interpreted as ```string``` by DBAL. I wasn't able to get this running, even without the ```isColumnNullable()``` method, so it might be that Laravel does that by default nowadays.

This change makes it that no results are never displayed in the Select, so this basically breaks the Enum field functionality for me.

**What did you use to test this? I assume it was something else than MySQL? If not, what version of MySQL did you run?**

I would love for this to be more general (use DBAL instead of SQL), but I don't think it's possible. As far as I remember, the ```enum``` field is only meant to be used with MySQL for a reason. Bad thing is - I don't remember that reason :-)) I think it's limited support for ENUM in other DBMS like SQLite & Postgre.

Reply:
this is the error from postgres... cuz `show columns` doesn't exist for PG.
I don't know what you mean by "didn't work". if you could fix whatever the MySQL problem is with those Eloquent commands? or show the error?

```SQLSTATE[42601]: Syntax error: 7 ERROR: syntax error at or near "FROM" LINE 1: SHOW COLUMNS FROM `articles` WHERE Field = "status" ^ (SQL: SHOW COLUMNS FROM `articles` WHERE Field = "status")```

Reply:
oh as far as dbal translating enum to a string.. it's more just that enum does not exist in dbal. it really shouldn't exist anywhere... you should of course maintain support for it ( and everything )... but it's a super limited/problematic/inefficient structure

Reply:
Sorry for not being clear. With this fix, in MySQL my "select" no longer shows any options. Previously, the user was able to pick from the options defined in the database column type, but after this fix they cannot do that anymore, since no options are shown. So this makes it more general, yes, but it also breaks the field for the only DBMS that was supported.

I agree with you that it should work on all DBMS. Theoretically :-) But **since Postgre doesn't support the ENUM column type, our ```enum``` field type can't  work on Postgre**. Developers on postgre should use [the ```select_from_array``` field type](https://laravel-backpack.readme.io/docs/crud-fields#section-select_from_array) instead, which pulls the options from PHP, not DB.

I've checked, and [we do have a notice in the documentation](https://laravel-backpack.readme.io/docs/crud-fields#section-enum) that this field type only works for MySQL databases.

Since:
- this field is only meant to be used with MySQL
- an error shows up only when this field is used without MySQL
- making the error go away (this fix) would make the field stop working on MySQL

I'm afraid I can't see how I can merge this PR. I guess I'd rather have an error thrown on Postgre and SQLite, so that users know they're not supposed to use this field type with those DBMS.

Reply:
Hmm… I think you do have a point @gareth-ib - it’s the demo that adds the ENUM field, so it’s the demo that’s the problem. We should NOT use a field that’s MySQL-specific inside a broad demo. I just added [an issue on the Demo](https://github.com/Laravel-Backpack/Demo/issues/68) for this - will get to tackle it soon.

Cheers!